src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -306,6 +306,39 @@ struct sdm_sensor_info
   } //get()
 };
 
+struct ert_status
+{
+  using result_type = query::ert_status::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type)
+  {
+    using tokenizer = boost::tokenizer< boost::char_separator<char> >;
+    auto pdev = get_pcidev(device);
+    std::vector<std::string> status;
+    result_type ertStatus = {0};
+    std::string errmsg;
+
+    pdev->sysfs_get("ert_ctrl", "status", errmsg, status);
+    if (!errmsg.empty())
+      throw xrt_core::query::sysfs_error(errmsg);
+
+    for (auto& line : status) {
+      // Format on each line: "<name>: <value>"
+      boost::char_separator<char> sep(":");
+      tokenizer tokens(line, sep);
+
+      auto tok_it = tokens.begin();
+      if (line.find("Connected:") != std::string::npos) {
+        ertStatus.connected = std::stoi(std::string(*(++tok_it)));
+        std::cout << line << std::endl;
+      }
+    }
+
+    return ertStatus;
+  }
```

Comment:
Similar to some other recent query addition, I don't think this scales,  the query_requests should be as simple as possible gathering the raw data from sysfs using emplace_sysfs_get, parsing of the raw data should be done upstream in query_requests.cpp.     Is that possible?

Reply:
I will update this PR and respect the rules.